src/analyses/invariant_set.cpp

```diff
@@ -1134,7 +1137,9 @@ void invariant_sett::apply_code(const codet &code)
   }
   else
   {
-    std::cerr << code.pretty() << '\n';
-    throw "invariant_sett: unexpected statement: "+id2string(statement);
+    INVARIANT(
```

Comment:
UNHANDLED_CASE seems more fitting than INVARIANT.  When we have some idea of what the program can contain during this analysis we may be able to say more.